docs/cloud9.md

```diff
@@ -0,0 +1,327 @@
+
+[**`[Go to tl;dr]`**](#tldr)
+
+## Contents
+
+  - [Context](#context)
+  - [Binstub versions](#binstub-versions)
+  - [Quick solution](#quick-solution)
+  - [Flexible solution](#flexible-solution)
+  - [Sources](#sources)
+  - [Versions](#versions)
+  - [tl;dr](#tldr)
+
+## Context
+
+This article describes how to properly configure
+[`webpack-dev-server`] with [`webpacker`] gem on a [Cloud9] workspace.
+After a preliminary remark about the proper binstub version of the
+`./bin/webpack-dev-server` script, this article presents two ways to
+tackle the task: a simple and [quick solution](#quick-solution), which
+is sufficient if we work alone on a project, and a slightly more
+involved but [flexible approach](#flexible-solution), that can be
+useful when several people might work in the same codebase.
+
+[Cloud9]: https://c9.io
+[`webpack-dev-server`]: https://github.com/webpack/webpack-dev-server
+[`webpacker`]: https://github.com/rails/webpacker
+
+## Binstub versions
+
+A lot of the confusion about the [`webpack-dev-server`] options and
+why they might not be properly taken into account, might be due to an
+outdated version of the `./bin/webpack-dev-server` script. The script
+created by the `rails webpacker:install` task of the [`webpacker`] gem
+v3.0.1 ([source][v3.0.1/lib/install/bin/webpack-dev-server.tt]) is not
+compatible with how v3.0.2 (sic) of the gem handles the
+[`webpack-dev-server`] option flags (see full list of
+[versions](#versions) below), which logically expects the
+corresponding [binstub version][#833] of the script
+([source][v3.0.2/exe/webpack-dev-server]). So please make sure that
+you are using the [correct binstub][v3.0.2/exe/webpack-dev-server]
+(the same applies to [`./bin/webpack`][v3.0.2/exe/webpack]). To be
+fair, the [changelog of v3.0.2] properly mentions the change:
+
+> - Added: Binstubs [#833]
+> - (...)
+> - Removed: Inline CLI args for dev server binstub, use env variables
+  instead
+
+[changelog of v3.0.2]: https://github.com/rails/webpacker/blob/v3.0.2/CHANGELOG.md#302---2017-10-04
+[v3.0.1/lib/install/bin/webpack-dev-server.tt]: https://github.com/rails/webpacker/blob/v3.0.1/lib/install/bin/webpack-dev-server.tt
+[v3.0.2/exe/webpack-dev-server]: https://github.com/rails/webpacker/blob/v3.0.2/exe/webpack-dev-server
+[v3.0.2/exe/webpack]: https://github.com/rails/webpacker/blob/v3.0.2/exe/webpack
+[#833]: https://github.com/rails/webpacker/pull/833/files
+
+## Quick solution
+
+If you are working alone, the easiest way to fix the configuration of
+the [`webpack-dev-server`] is to add an entry to the `/etc/hosts` and
+to modify the `development.dev_server` entry of the
+`config/webpacker.yml` file.
+
+### `/etc/hosts` entry
+
+As mentioned in some of the [sources](#sources) below, you have to add
+an entry to [`/etc/hosts`](https://askubuntu.com/a/183187) to alias
+the hostname of your [Cloud9] workspace to `0.0.0.0`:
+
+```bash
+echo "0.0.0.0 ${C9_HOSTNAME}" | sudo tee -a /etc/hosts
```

Comment:
Hi, thanks for putting this article together!

I think this step is not necessary if you keep `host: localhost` in `webpacker.yml` (line 111 in this PR). Note that setting `public:` (line 113) to a value of `C9_HOSTNAME` is necessary.

Reply:
Thanks for pointing this out, @ssemakov! I've tested it in two of my Cloud9 workspaces and it works as you describe. I've updated the doc and the PR to reflect your suggestion. Cheers.